gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/experimental/uikit/internal/configureInstallIosDeployTask.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020-2022 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.compose.experimental.uikit.internal
+
+import org.gradle.api.Project
+import org.jetbrains.compose.desktop.application.internal.MacUtils
+import org.jetbrains.compose.desktop.application.internal.UnixUtils
+import org.jetbrains.compose.experimental.uikit.tasks.AbstractComposeIosTask
+
+internal const val TASK_INSTALL_IOS_DEPLOY = "iosInstallIosDeploy"
+internal val Project.iosDeployExecutable get() = iosDeploySrc.resolve("build/Release/ios-deploy")
+
+private const val IOS_DEPLOY_GIT = "https://github.com/ios-control/ios-deploy.git"
+private const val IOS_DEPLOY_TAG = "1.11.4"
+private val Project.iosDeploySrc get() = rootProject.buildDir.resolve("ios-deploy-$IOS_DEPLOY_TAG-src")
+
+internal fun Project.configureInstallIosDeployTask() {
+    tasks.composeIosTask<AbstractComposeIosTask>(TASK_INSTALL_IOS_DEPLOY) {
+        onlyIf { !iosDeployExecutable.exists() }
+        doLast {
+            iosDeploySrc.deleteRecursively()
```

Comment:
Not sure if this is a good idea to delete this way - things may go terribly wrong.

Reply:
If the ios-deploy utility installs successfully, then this step will be skipped. The condition is responsible for this: 
`onlyIf { !iosDeployExecutable.exists() }`

On the other side:
If an error occurs during git clone or build. Then at the next attempt - the directory will be deleted and the process will start from the very beginning.

Reply:
Let's keep it this way then.